cli/command/image/tree.go

```diff
@@ -0,0 +1,296 @@
+package image
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/docker/cli/cli/command"
+	"github.com/morikuni/aec"
+
+	"github.com/containerd/platforms"
+	"github.com/docker/docker/api/types/filters"
+	imagetypes "github.com/docker/docker/api/types/image"
+	"github.com/docker/docker/pkg/stringid"
+	"github.com/docker/go-units"
+)
+
+type treeOptions struct {
+	all     bool
+	filters filters.Args
+}
+
+func runTree(ctx context.Context, dockerCLI command.Cli, opts treeOptions) error {
+	images, err := dockerCLI.Client().ImageList(ctx, imagetypes.ListOptions{
+		All:            opts.all,
+		ContainerCount: true,
+		Filters:        opts.filters,
+		Manifests:      true,
+	})
+	if err != nil {
+		return err
+	}
+
+	view := make([]topImage, 0, len(images))
+	for _, img := range images {
+		details := imageDetails{
+			ID:        img.ID,
+			DiskUsage: units.HumanSizeWithPrecision(float64(img.Size), 3),
+			Used:      img.Containers > 0,
+		}
+
+		children := make([]subImage, 0, len(img.Manifests))
+		for _, im := range img.Manifests {
+			if im.Kind != imagetypes.ImageManifestKindImage {
+				continue
+			}
+
+			imgData := im.ImageData
+			platform := imgData.Platform
+
+			sub := subImage{
+				Platform:  platforms.Format(platform),
+				Available: im.Available,
+				Details: imageDetails{
+					ID:        im.ID,
+					DiskUsage: units.HumanSizeWithPrecision(float64(im.Size.Total), 3),
+					Used:      len(imgData.Containers) > 0,
+				},
+			}
+
+			children = append(children, sub)
+		}
+
+		view = append(view, topImage{
+			Names:    img.RepoTags,
+			Details:  details,
+			Children: children,
+		})
+	}
+
+	return printImageTree(dockerCLI, view)
+}
+
+type imageDetails struct {
+	ID        string
+	DiskUsage string
+	Used      bool
+}
+
+type topImage struct {
+	Names    []string
+	Details  imageDetails
+	Children []subImage
+}
+
+type subImage struct {
+	Platform  string
+	Available bool
+	Details   imageDetails
+}
+
+const headerSpacing = 3
+
+//nolint:gocyclo
+func printImageTree(dockerCLI command.Cli, images []topImage) error {
+	out := dockerCLI.Out()
+	_, width := out.GetTtySize()
+	if width == 0 {
+		width = 80
+	}
+	if width < 20 {
+		width = 20
+	}
+
+	headers := []header{
+		{Title: "Image", Width: 0, Left: true},
+		{Title: "ID", Width: 12},
+		{Title: "Disk use", Width: 8},
+		{Title: "Used", Width: 4},
```

Comment:
wdyt?

I prefer "Size" over "Disk use", especially when next to "Used" (or "In use").

```suggestion
		{Title: "Name", Width: 0, Left: true},
		{Title: "ID", Width: 12},
		{Title: "Size", Width: 8},
		{Title: "In use", Width: 6},
```

Reply:
Size is more ambiguous though - it could mean just the size of the image (e. g. when pushing).
`Disk use` here is the total space occupied on the daemon host both by the distributable images blobs AND the image snapshots (content needed to run containers off this image).

Also, I picked `Used` because the value is either empty or a single tick (✔️), so prefer to keep the column name short to not use too much space.

Reply:
If we want to keep "Disk use", I would change it to "Disk usage" (sorry, also longer).

I "Used" referring to running containers, or also stopped containers? "Used" is more ambiguous than "In use" w.r.t. to the current state (is it currently in use? has it been used but isn't anymore?). When I initially proposed "In use" I was thinking currently running containers, but I guess it's probably also for stopped containers, in which case "Used" is fine.

Reply:
Stopped containers also count.

Reply:
Changed `Disk use` to `Disk usage`